src/sql-parser/src/ast/defs/ddl.rs

```diff
@@ -458,13 +458,31 @@ impl_display!(DbzMode);
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash, EnumKind)]
 #[enum_kind(ConnectorType)]
-pub enum CreateConnector<T: AstInfo> {
+pub enum CreateConnector {
     Kafka {
         broker: String,
-        with_options: Vec<SqlOption<T>>,
+        with_options: Vec<WithOption>,
     },
 }
 
+impl AstDisplay for CreateConnector {
+    fn fmt<W: fmt::Write>(&self, f: &mut AstFormatter<W>) {
+        match self {
+            Self::Kafka {
+                broker,
+                with_options,
+            } => {
+                f.write_str("FOR KAFKA BROKER '");
```

Comment:
The `AstDisplay` impl for a given type should include only the syntax that directly corresponds to that type:

```suggestion
                f.write_str("KAFKA BROKER '");
```